lib/thrift/parser/resolver.ex

```diff
@@ -30,6 +30,16 @@ defmodule Thrift.Parser.Resolver do
       end
     end)
 
-    Map.merge(state, new_mappings)
+    existing_keys = Map.keys(state) |> MapSet.new
+    new_keys = Map.keys(new_mappings) |> MapSet.new
+    shared_keys = MapSet.intersection(existing_keys, new_keys) |> MapSet.to_list
```

Comment:
Do we need to convert this MapSet to a list? It looks like the following condition is just a emptiness test, which you can perform using `MapSet.size/1`.